### PR TITLE
Fix classifier bug

### PR DESCRIPTION
**Pull request by ahn1340**
This PR fixes the problem of the ensemble of classifiers returning prediction values larger than 1. This happens because predict() in ensemble_selection.py sometimes receive model predictions as the input, where predictions of models with zero weights are excluded, and sometimes it receives predictions including those of zero weight models. Therefore, predict() now deals with these two cases to make sure that model weights are applied correctly.

### CODE REVIEW

File: autosklearn/ensemble_builder.py [hunk 1]
```diff
@@ -210,17 +210,17 @@ def main(self):
             
             # train ensemble
             ensemble = self.fit_ensemble(selected_keys=selected_models)
-            
+
```
Comment (mfeurer) [inline autosklearn/ensemble_builder.py]: Please revert changes in this file as there are no changes to the actual code in this file.

File: autosklearn/ensembles/ensemble_selection.py [hunk 2]
```diff
@@ -194,8 +194,10 @@ def _bagging(self, predictions, labels, fraction=0.5, n_bags=20):
         return np.array(order_of_each_bag)
 
     def predict(self, predictions):
-        non_null_weights = (weight for  weight in self.weights_ if weight > 0)
-        for i, weight in enumerate(non_null_weights):
+        #non_null_weights = (weight for weight in self.weights_ if weight > 0)
```
Comment (mfeurer) [inline autosklearn/ensembles/ensemble_selection.py]: Could you please remove these comments?

File: autosklearn/ensembles/ensemble_selection.py [hunk 3]
```diff
@@ -194,8 +194,10 @@ def _bagging(self, predictions, labels, fraction=0.5, n_bags=20):
...
+        #for i, weight in enumerate(non_null_weights):
+        #    predictions[i] *= weight
+        for i, weight in enumerate(self.weights_):
             predictions[i] *= weight
         return np.sum(predictions, axis=0)
```
Comment (mfeurer) [inline autosklearn/ensembles/ensemble_selection.py]: Could this code maybe be simplified a lot more to `return np.average(predictions, axis=0, weights=self.weights_)`?

File: autosklearn/estimators.py [hunk 4]
```diff
@@ -527,8 +528,12 @@ def predict_proba(self, X, batch_size=None, n_jobs=1):
             The predicted class probabilities.
 
         """
-        return super().predict_proba(
+        pred_proba = super().predict_proba(
             X, batch_size=batch_size, n_jobs=n_jobs)
+        assert np.allclose(np.sum(pred_proba, axis=1),
+                           np.ones_like(pred_proba[:, 0])),\
+        "prediction probability does not sum up to 1!"
```
Comment (mfeurer) [inline autosklearn/estimators.py]: Could you please change the formatting to
```
assert (
    np.allclose(
        np.sum(pred_proba, axis=1),
        np.ones_like(pred_proba[:, 0])),
    ), "prediction probability does not sum up to 1!"

Comment (ahn1340) [reply autosklearn/estimators.py]: Yep, I will make that modification. Currently some unittests fail due to this line. I'm working on the fix and I will remove [WIP] as soon as it is done.

**Pull request merged**